src/BitSubAccount.ts

```diff
@@ -9,13 +9,17 @@ import {
   TxsSignedOrUnSigned
 } from './fetchers/SubAccountAPI'
 import { BitErrorCode, DotbitError } from './errors/DotbitError'
+import { isSubAccount } from './tools/account'
 
 export class BitSubAccount extends BitAccount {
   isSubAccount = true
   mainAccount: string
 
   constructor (options: BitAccountOptions) {
     super(options)
+    if (!isSubAccount(options.account)) {
```

Comment:
InvalidAccountId is used for accountId, which is 32 bytes hex string.
You can add a new ErrorCode.